### ISSUE TITLE
Template rendering support in Phoenix v1.7+

### ISSUE THREAD

**Issue by cbortz** [appsignal/appsignal-elixir-phoenix#52]
Release candidates for Phoenix v1.7 have recently rolled out, and I'm experimenting with using it for a new project. There are a lot of changes, one of them being `Phoenix.View` not being used in the web module. As a result, there isn't a `view/0` either. It appears to have been replaced by `html/0`. I figured I'd attempt to instrument anyway, adapting the implementation in the [docs for Template Rendering](https://docs.appsignal.com/elixir/integrations/phoenix.html#template-rendering) by injecting the following line into the `html/0` function in the web module:

```elixir
use Appsignal.Phoenix.View
```

When starting the app, the following error occurs:

```
== Compilation error in file lib/web/controllers/error_html.ex ==
** (CompileError) lib/web/controllers/error_html.ex:1: undefined function __templates__/0 (expected Web.ErrorHTML to define such a function or for it to be imported, but none are available)
```

This was expected for the most part, but was hoping otherwise :sweat_smile:. That being said, have y'all started investigating out of the box support for Phoenix 1.7? Do you have any recommendations for handling this in the meantime?

---

For reference, I am using Phoenix v1.7.0-rc.0.

**Comment by jeffkreeftmeijer** [comment]
Hey @cbortz,

Thanks for creating this issue. You’re right, the `Appsignal.View` solution doesn’t work on the new Phoenix release candidate. We’re looking into how we can restore this functionality for 1.7, and we’ll keep you posted here.

**Comment by jeffkreeftmeijer** [comment]
Quick update; due to changes in the way Phoenix sets up its views in the new version, our previous approach is no longer viable. However, I found a way to get instrumentation for template rendering to work through :telemetry. I’ve [asked the Phoenix maintainers](https://github.com/phoenixframework/phoenix_template/pull/4) for input on a patch needed for that.

**Comment by cbortz** [comment]
@jeffkreeftmeijer Thanks for the update and for looking into this! 🙏

**Comment by jeffkreeftmeijer** [comment]
One thing I forgot to mention (for you and others who might find this through search); for running Phoenix 1.7 right now, you can safely remove the `use Phoenix.View` line from your app. You won’t get view instrumentation, but the rest of AppSignal should work as expected on 1.7.

**Comment by jeffkreeftmeijer** [comment]
@cbortz https://github.com/phoenixframework/phoenix/commit/f095d1f82cdf18ffc69226fd59ce36a7fc2ffb44 was accepted into Phoenix, and  I’ve added a listener for it in the https://github.com/appsignal/appsignal-elixir-phoenix/tree/template_telemetry branch. If you have time, could you update your app to the latest Phoenix main, and check out appsignal_phoenix’s template_telemetry branch, to see how it works in your app?

**Comment by cbortz** [comment]
@jeffkreeftmeijer Sorry for the delayed response! We can likely test this out within the next two weeks. We'd shifted focus on a different part of the app, but will get back to templates soon. 🤞

**Comment by tombruijn** [comment]
This has been shipped in version 2.2.0 of appsignal_phoenix package. Please try it out!

**Comment by cbortz** [comment]
Quick follow-up... I am also receiving this error with appsignal enabled.

**Comment by cbortz** [comment]
Thanks @jeffkreeftmeijer I just tested with that branch and everything looks great, both when AppSignal is enabled and disabled.

**Comment by jeffkreeftmeijer** [comment]
@cbortz thanks for checking. I’ve just released this patch as AppSignal for Phoenix 2.2.1, so you should be able to switch to that.
